added documentation of platform maintenance (#481)
meshcloud/meshcloud-docs: docs/administration.platforms.md

---
id: administration.platforms
title: meshPlatforms
---

> A key task for Platform Operators is integrating your Cloud Platforms as meshPlatforms into meshStack. This enables

Reviewer: "as meshPlatforms" -> I would just add a short explanation on what that means. I could imagine first time users being a bit confused by the terminology.

Author: added a sentence before

> tenant creation, replication and metering of your Cloud Platforms. The maintenance includes

Reviewer: ```suggestion
the core functionality of meshStack such as tenant creation, replication, and metering of your Cloud Platforms. The maintenance includes
```

> general information about the Cloud Platform as well as structuring and configuring them.

Reviewer: I don't really get this sentence: "The maintenance [...] configuring them". What are you trying to convey here?

> A central part of it also includes the application of security and governance which is covered in the

Reviewer: The two points you mention about landing zones and tenants would prob. go better in a bullet list. something like:

```
Besides integrating your cloud platform, the following functionality is also highly useful:
* Manage and create landing zones in the [landing zone administration screen](./linkto)
* View all your existing cloud tenants in the [tenant list](./linkto)
```
I think this also bridges better to your point of this functionality becoming a part of the platform control plane in future.

[Landing Zone Administration](administration.landing-zones.md).

> Getting an insight into all tenants that are being used in a certain Cloud Platform can be found in the

Author: replaced it with jelle's proposed text

[Tenants](administration.tenants.md) section.

> All platform related maintenance like Landing Zones, Tenants, etc will be moved to the [Platform Control Plane](#platform-control-plane)
> in future.

## Platform Creation

You can integrate new meshPlatforms in self-service via a wizard in meshStack.

![Platform Creation](assets/platform_maintenance/platform-creation.png)

The following platform types are available for selection.

![Platform Types](assets/platform_maintenance/platform-types.png)

> You have to assign a meshLocation to your meshPlatform and can provide additional provider information to the meshLocation if you

Reviewer: I would recommend to add some best practices for the meshLocation. I could imagine that users do not really know what to do with them initially. Something like:

The meshLocation is useful for grouping together multiple platforms. This is especially helpful for private cloud platforms that have an actual physical location. For public cloud platform, we recommend to name them according to the provider as public cloud platforms are not really tied to a physical location. E.g. name the meshLocation for Azure simply "Azure".

> You have to assign a meshLocation to your meshPlatform and can provide additional provider information to the meshLocation if you

Reviewer 2: I would rephrase the part

> You have to assign a meshLocation to your meshPlatform and can provide additional provider information [...]

probably into two sentences, because one is a requirement and one is an option.

Author: changed this sentence anyway to mainly link to the dedicated meshLocation section

> would like to. This is the same information you can provide in the [Settings](#settings) tab of the Platform Control Plane.

Reviewer: ```suggestion
would like to. You can always change this information later in the [Settings](#settings) tab of the Platform Control Plane.
```

Author: moved this to the end of the Platform Creation section as the restriction part is also covered in the settings tab

Additionally you can directly [restrict](#restrict-platform-access) your meshPlatform to certain meshCustomers to e.g. first test a new meshPlatform integration

> with a meshCustomer you have access to. This can prevent confusion of other users who may see the meshPlatform otherwise

Reviewer: ```suggestion
with a private meshCustomer that only you have access to. This can prevent confusion of other users who may see the meshPlatform otherwise
```

> and would try to use it eventhough it is not working yet. Or you just have a dedicated Cloud Platform that shall only be available

Reviewer 2: ```suggestion
and would try to use it eventhough it is not working yet. Another use case is to have a dedicated Cloud Platform that shall only be available
```

to certain meshCustomers in general.

## Platform Control Plane

> The Platform Control Plane is the central entry point for any tasks or insights you want to do or get about your meshPlatform.

Reviewer 2: ```suggestion
The Platform Control Plane is the central entry point for performing any tasks on your meshPlatform or insights you want to gather on it.
```

![Platform Control Plane](assets/platform_maintenance/control-plane.png)

### Financials

In this tab you can get an overview about which kind of resources are being used in your Cloud Platform.

### Settings

This tab is mainly responsible for managing how your meshPlatform is shown and used inside meshStack. You can provide general information
about your Platform like a display name or the Web Console Url meshStack will link to when the user wants to actually access the cloud platform.

> 

Reviewer: Maybe show a tip or smth: "For public cloud platforms you can leave the Web Console Url as is".

> You can also group your meshPlatforms into meshLocations to e.g. combine multiple private cloud platforms that are running in the same data center

Reviewer: ah this is the point that I made before. Maybe you can move it up?

> into one meshLocation. Or you are using e.g. multiple Azure Tenants for your different stages and want to group them into an Azure meshLocation.

Reviewer 2: > Or you are using e.g. multiple Azure Tenants for [..]

In general I would avoid sentences that start with and / or

> Additionally you can provide some Provider information of your meshLocation. You can describe how you operate your Cloud Platforms that belong to a meshLocation,

Reviewer 2: I found `provider information` and `Provider information`, please check again that capitalization is the same every time.

you can provide a link to a Wiki to provide further, more detailed information about your meshPlatforms within a meshLocation.

#### Restrict Platform Access

> Access to a specific platform can be restricted via the "Restrictions" tab. This feature is helpful

Reviewer: This was already mentioned before. I would suggest to structure the document in a way that you won't have to repeat yourself for the creation and updating part of things.

when a new meshPlatform shall be integrated, but initially only be visible to a few customers for integration testing.
Or you have a dedicated Cloud Platform that shall only be available to certain meshCustomers in general.

To restrict the meshPlatform, search for the customers the platform shall be restricted to.

> Add these customers via the *+* button.

Reviewer 2: maybe connect the two senctences with and?

> To restrict the meshPlatform, search for the customers the platform shall be restricted to and add these customers via the *+* button.

> Currently only OpenShift is supported for Quota Management

![Manage Platform Quota Definitions](assets/tenants/platform-quota-definitions.png)

### Platform Deactivation

Via the "Danger Zone" tab you can deactivate or reactivate your meshPlatform.

> Deactivation will make the meshPlatform not available anymore to new meshTenants, existing meshTenants won't be accessible via meshPanel anymore and no more replication will be executed. Metering instead will still be executed for non-deleted meshTenants of this meshPlatform for now. This will change in future. Direct access in the cloud platform itself to the platform tenants is still possible as meshStack does not delete or do any updates on platform tenants when a meshPlatform is deactivated.

Reviewer: "This will change in future". This sounds a bit vague, can we by any chance already provide more info on the topic or even get rid of this sentence altogether?

Reviewer 2: I'm in favor of getting rid of it and just adapt the description when the behavior is changed,

> Deactivation will make the meshPlatform not available anymore to new meshTenants, existing meshTenants won't be accessible via meshPanel anymore and no more replication will be executed. Metering instead will still be executed for non-deleted meshTenants of this meshPlatform for now. This will change in future. Direct access in the cloud platform itself to the platform tenants is still possible as meshStack does not delete or do any updates on platform tenants when a meshPlatform is deactivated.

Reviewer 2: i'd use unavailable instead of "not available"

> Deactivation will make the meshPlatform not available anymore to new meshTenants, existing meshTenants won't be accessible via meshPanel anymore and no more replication will be executed. Metering instead will still be executed for non-deleted meshTenants of this meshPlatform for now. This will change in future. Direct access in the cloud platform itself to the platform tenants is still possible as meshStack does not delete or do any updates on platform tenants when a meshPlatform is deactivated.

Reviewer 2: maybe "platform's tenants" is a bit clearer then "platform tenants" ?

notifications:

- **Platforms**: Select the platforms relevant to your notification so they can be shown on the affected platform's dashboards. Users
  can also [subscribe](meshcloud.profile.md#profile) to notifications by platform and [subscribed users](./meshcloud.profile.md#platform-notification-subscriptions) will receive new notifications via email.
- **Message**: The message that will be shown to all users. You can enter valid HTML in this field (except for the `<p>` tag). This is useful for e.g. supplying links to support pages.
- **Severity**: The type of message (info, warning, critical).
  - Info: for new features, upcoming services updates and general information that is not expected to disrupt platform operation
  - Warning: best used to inform about partial platform outages or problems that may occur for some users
  - Critical: appropriate in case of complete platform failures or impending platform downtimes
- **Show From**: The earliest time the message will be shown. You can use this to create messages that will only be shown on at a later date.

> - **Show Until**: After this time the message will no longer be displayed.

Reviewer: For these two values there is also behavior when leaving them empty right? Maybe describe that too.
